README.md

```diff
@@ -67,19 +70,26 @@ $ cd hyperopt
 ```
 
 ### 4. Create a virtual env (python 3.x) to contain the dependencies and activate it
-on Windows using virtualenv:  
-`$ python -m venv venv`  
-or with conda  
-`$ conda create --name hyperopt`  
-or Linux/Unix  
-`$ python3 -m venv venv`
+Create virtualenv with:  
+`$ python -m venv my_env` or `$ python3 -m venv my_env`  
+or with conda:  
+`$ conda create -n my_env python=3`
 
+Activate on Linux/Unix:  
+`$ source my_env/bin/activate`  
+or with conda:  
+`$ conda activate my_env`
+
+then install:
 ```bash
-$ source venv/bin/activate
-$ python setup.py &&  pip install -e
-'.[MongoTrials, SparkTrials, ATPE, dev]'`
+$ python setup.py develop
+$ pip install -e .[MongoTrials]
+$ pip install -e .[SparkTrials]
+$ pip install -e .[ATPE]
+$ pip install -e .[dev]
```

Comment:
this is not needed, please change to `pip install -e '.....'` only

Reply:
have you tested your assumption? I found that you cannot install multiple extras in one command using pip. The only way to guarantee the extras install is to use a single line. Tried the following:

```bash
pip install -e '.[MongoTrials, SparkTrials, ATPE, dev]'
pip install -e .[MongoTrials, SparkTrials, ATPE, dev]
pip install -e .["MongoTrials", "SparkTrials", "ATPE", "dev"]
pip install -e '.["MongoTrials", "SparkTrials", "ATPE", "dev"]'
```

of which none work consistently.

Reply:
this worked for me `pip install -e '.[ATPE, dev]'`. I just tested. I can imagine that with 4 should work as well. Can you test:
1. create a fresh environment
2. do `pip install -e '.[ATPE, dev]'`
3. check that black and lightgbm are installed (they come each with one of those extras)

If this doesn't work might be because of Windows... but seems weird to me tbh

Reply:
> 
> 
> > LGTM. Added just some more comments to the README. Please make sure that the tests pass. Unfortunately some our tests are not stable (very welcome to fix), so you might need to commit again to get a proper seed and pass the tests. You can run the tests locally first to make sure everything works before using the CI
> 
> make sure tests pass @cibic89

This test fails on both Windows and Ubuntu with the same error:
```
hyperopt/tests/test_atpe_basic.py Fatal Python error: Aborted

Thread 0x00007fad58bab700 (most recent call first):
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 300 in wait
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 552 in wait
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/tqdm-4.48.0-py3.7.egg/tqdm/_monitor.py", line 69 in run
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 926 in _bootstrap_inner
  File "/home/george/miniconda3/lib/python3.7/threading.py", line 890 in _bootstrap

Current thread 0x00007fad62221740 (most recent call first):
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/lightgbm/basic.py", line 1744 in __init__
  File "/home/george/code_projects/hyperopt/hyperopt/atpe.py", line 682 in __init__
  File "/home/george/code_projects/hyperopt/hyperopt/atpe.py", line 1575 in suggest
  File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 269 in run
  File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 352 in exhaust
  File "/home/george/code_projects/hyperopt/hyperopt/fmin.py", line 539 in fmin
  File "/home/george/code_projects/hyperopt/hyperopt/tests/test_atpe_basic.py", line 28 in test_run_basic_search
  File "/home/george/miniconda3/lib/python3.7/unittest/case.py", line 628 in run
  File "/home/george/miniconda3/lib/python3.7/unittest/case.py", line 676 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/unittest.py", line 231 in runtest
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 135 in pytest_runtest_call
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 217 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 244 in from_call
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 217 in call_runtest_hook
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 186 in call_and_report
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 100 in runtestprotocol
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/runner.py", line 85 in pytest_runtest_protocol
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 272 in pytest_runtestloop
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 247 in _main
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 191 in wrap_session
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/main.py", line 240 in pytest_cmdline_main
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/george/code_projects/hyperopt/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 125 in main
  File "/home/george/code_projects/hyperopt/venv/bin/pytest", line 8 in <module>
Aborted (core dumped)

```

Reply:
> 
> 
> this worked for me `pip install -e '.[ATPE, dev]'`. I just tested. I can imagine that with 4 should work as well. Can you test:
> 
>     1. create a fresh environment
> 
>     2. do `pip install -e '.[ATPE, dev]'`
> 
>     3. check that black and lightgbm are installed (they come each with one of those extras)
> 
> 
> If this doesn't work might be because of Windows... but seems weird to me tbh

You are spot on; it was failing because of Windows but don't have this issue any more. Switched to Ubuntu.

Reply:
we should give instructions that work on Windows as well. So maybe do two (sets of) commands: one for Windows users, and one for the rest, as for the virtualenv. Eventually we can revisit this to try to generate instructions that are short and work for both...

Reply:
> 
> 
> we should give instructions that work on Windows as well. So maybe do two (sets of) commands: one for Windows users, and one for the rest, as for the virtualenv. Eventually we can revisit this to try to generate instructions that are short and work for both...

The single line method installs the extras on Windows but I suspect there are other problems with pytest on this platform. Let's push this for another time I think.